pyHalo/realization_extensions.py

```diff
@@ -497,3 +502,212 @@ def _get_fluctuation_halos(realization, fluctuation_amplitude, fluctuation_size,
                              unique_tag=np.random.rand()) for i in range(len(amps))]
 
     return fluctuations
+
+def mask_annular(center_x, center_y, x_grid, y_grid, r_min, r_max = None):
+    """
+    An annular mask of specified inner and outer radii
+    
+    :param center_x: x-coordinate of center position of circular mask
+    :param center_y: y-coordinate of center position of circular mask
+    :param x_grid: x-coordinate grid
+    :param y_grid: y-coordinate grid
+    :param r_min: inner radius of mask in units of grid coordinates
+    :param r_max: outer radius of mask in units of grid coordinates
+    :return: mask array of shape x_grid with =0 inside the radius and =1 outside
+    """
+    x_shift = x_grid - center_x
+    y_shift = y_grid - center_y
+    R = np.sqrt(x_shift*x_shift + y_shift*y_shift)
+    mask = np.ones(x_shift.shape)
+    if r_max == None:
+        mask[R <= r_min] = 0
+    else:
+        mask[R <= r_min] = 0
+        mask[R >= r_max] = 0
+
+    return mask
+
+def convergence_map(realization, map_size, npix, lens_model_list_macro=None, kwargs_lens_macro=None, redshift_list_macro=None):
```

Comment:
It looks like the only difference between this function and the multiplane_convergence function is the setup of the XX and YY grids, but these grids aren't actually used by this function so I don't think they belong here. To avoid duplicating the functionality of multiplane_convergence, can we instead just pass in map_size to the corr_kappa_with_mask function (instead of XX_ and YY_), and grid the grids there?